applications/auto_setup/backend/setup_cavity.py

```diff
@@ -36,9 +36,30 @@ def __init__(
         self.note_pv: str = self.auto_pv_addr("NOTE")
         self._note_pv_obj: Optional[PV] = None
 
+        if self.number is not None:
```

Comment:
You shouldn't need this check either, I'd just pull everything out of the if statement